### PR TITLE
Restructuring README and docs

### PR DESCRIPTION
**Pull request by sharonsolomon**
#1728 Feature Request: Split readme into more actionable chunks

Description

This is a resubmit of a now-closed pull request.  In this version, all conflicts with 8.1 have been resolved. I have also updated the sidebar.

NOTE: When the microsite goes live the link to it in the main README.md will need to be changed as will the first link of HOWTO_CODE_REVIEW.md. Those links currently point to the docs folder.

The goal of this PR is to make the documentation more beginner-friendly and easier to navigate for everyone. The idea is to funnel 'quick start' users to only the information they need and funnel more advanced users to the microsite and developer deployment instructions.

The PR moves several sections of the readme to their own documents. It removes the list of links at the bottom in favor of a link to the new microsite. The readme was then edited for flow and clarity.

The document about how to hire someone to install Spoke was also split up to remove the sections on deciding if spoke is right for you and deciding how to deploy Spoke. Those two sections are now their own docs linked to from the readme.

Other docs have been renamed to match the suggested conventions since all links would need to be revisited anyway. All internal and external links have been edited and checked.

I did not update any links in the Release Notes document. If this is needed, please let me know.

I split up the changes into quite a few commits with notes on each to keep each small enough for review.

### CODE REVIEW

File: docs/HOWTO_CODE_REVIEW.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
 # How to Code Review
-Thank you for stepping up to code review changes for Spoke! Check out [our developer docs](https://github.com/MoveOnOrg/Spoke#more-documentation) for more information about how we write code.
+Thank you for stepping up to code review changes for Spoke! Check out [our developer docs](https://github.com/MoveOnOrg/Spoke/tree/main/docs) for more information about how we write code.
```
Comment (ibrand) [inline docs/HOWTO_CODE_REVIEW.md]: Let's use `EXPLANATION-development-guidelines.md` even if it's a little messy of a doc

File: docs/HOWTO_DEVELOPMENT_LOCAL_SETUP.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+## Getting started
+
+The following is the reccomended developer set-up.  If you prefer, [click here for instructions for a minimalist set-up.](HOWTO_MINIMALIST_DEPLOY.md)
```
Comment (ibrand) [inline docs/HOWTO_DEVELOPMENT_LOCAL_SETUP.md]: This minimalist set-up is a deployment method, so it's not really a replacement for setting up a development environment. I think we should link to it in the "additional ways to deploy" in the README

**Review by ibrand — CHANGES_REQUESTED** [review]
Thank you so much for all of these changes! I wrote a lot of comments, and please feel free to continue conversation on any of them.

File: docs/HOWTO_CODE_REVIEW.md (hunk 1 above)
Comment (sharonsolomon) [reply docs/HOWTO_CODE_REVIEW.md]: Agreed. The idea on this one is to replace it with a link to the developer section in the microsite when we have the URL.

File: docs/EXPLANATION_CHOOSE_A_SETUP.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+## Deciding how to run Spoke
+
+At a high level, you need to decide how much control you want over your deployment architecture. Control over system scaling issues and security trades off against amount of time it'll take to setup and maintain your system. 
+
+For guidance on hiring someone to
```
[the remark this answers is not in the document]
Comment (sharonsolomon) [reply docs/EXPLANATION_CHOOSE_A_SETUP.md]:
```suggestion

```

[the remark this answers is not in the document]
Comment (sharonsolomon) [reply docs/EXPLANATION_CHOOSE_A_SETUP.md]: Adding this link to the bottom of this section instead.

File: docs/EXPLANATION_CHOOSE_A_SETUP.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+1. You aren't yet sure if you are able to / want to scale up your system and just want to try it out first
+1. The cost of devops time spent implementing control over your deployment architecture isn't worth the benefit textbanking has to your organization
+
+There are many ways to deploy software onto hardware and many cloud computing providers. In this set of documentation, we use Heroku and Amazon AWS. AWS is more work to setup than Heroku, but gives you more control over scaling and security. In our experience, a full stack engineer with 2+ yrs experience that includes AWS can set this up in 5 days. There is also a Spoke open source community anchored in the Progressive Coders Network slack who can help with setup.
+
+If you're ok with less control, you can follow the instructions [here](HOWTO_HEROKU_DEPLOY.md) to deploy an instance of Spoke to Heroku  
+
+If you want more control, you can follow the instructions [here](HOWTO_DEPLOYING_AWS_LAMBDA.md) to deploy an instance of Spoke onto Amazon AWS. Many people use dedicated EC2 servers for deployment, but we've found lambda (pay-per-invocation vs hour) to be more cost effective for bursty traffic.
+
```
Comment (sharonsolomon) [inline docs/EXPLANATION_CHOOSE_A_SETUP.md:19]:
```suggestion

For guidance on hiring someone to set up Spoke for you, see [here.](HOWTO_HIRE_SOMEONE_TO_INSTALL_SPOKE.md)

```

File: docs/HOWTO_CODE_REVIEW.md (hunk 1 above)
Comment (sharonsolomon) [inline docs/HOWTO_CODE_REVIEW.md]:
```suggestion
Thank you for stepping up to code review changes for Spoke! Check out [our developer docs](EXPLANATION-development-guidelines.md) for more information about how we write code.
```

File: docs/HOWTO_DEVELOPMENT_LOCAL_SETUP.md (hunk 2 above)
Comment (sharonsolomon) [inline docs/HOWTO_DEVELOPMENT_LOCAL_SETUP.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -34,121 +34,46 @@ Please let us know if you deployed by filling out this form [here](https://act.m
...
-Docker is optional, but can help with a consistent development environment using postgres. You can also set up postgres without docker ([documented here](https://github.com/MoveOnOrg/Spoke/blob/main/docs/HOWTO_USE_POSTGRESQL.md)) but we recommend the docker route.
+For developers, please see our recomendations for [deploying locally for development](/docs/HOWTO_DEVELOPMENT_LOCAL_SETUP.md).
 
-1. Install docker and docker compose
 
-- Docker allows you to run apps in containers and can be installed [here with Docker's instructions](https://docs.docker.com/desktop/)
-- Docker Compose is the tool used to create and run docker configurations. If you installed Docker on Mac, you already have Docker Compose, if you're using Linux or Windows you can install Docker Compose [with these instructions](https://docs.docker.com/compose/install/)
 
-2. Run `./dev-tools/create-test-database` to populate the test database
+## Quick Start with Heroku
```
Comment (sharonsolomon) [inline README.md]:
```suggestion
### Quick Start with Heroku
```

File: README.md [hunk 6]
```diff
@@ -34,121 +34,46 @@ Please let us know if you deployed by filling out this form [here](https://act.m
...
-5. Once you've created your organization, we recommend setting the env var `SUPPRESS_SELF_INVITE=1` so you don't get prompted to create a new org every time you log in
-6. See the [Admin](https://youtu.be/PTMykMX8gII) and [Texter](https://youtu.be/EqE1UDvKGco) demos to learn about how Spoke works.
-7. See [Getting Started with Development](#more-documentation) below.
-8. See [How to Run Tests](https://github.com/MoveOnOrg/Spoke/blob/main/docs/HOWTO-run_tests.md)
+Please let us know if you deployed by filling out this form [here](https://act.moveon.org/survey/tech/)
 
-### SMS
 
-For development, you can set `DEFAULT_SERVICE=fakeservice` to skip using an SMS provider (Twilio or Nexmo) and insert the message directly into the database.
+## Other Options for Production Use
```
Comment (sharonsolomon) [inline README.md]:
```suggestion
### Other Options for Production Use 
```

File: docs/EXPLANATION_CHOOSE_A_SETUP.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+## Deciding how to run Spoke
+
```
Comment (sharonsolomon) [inline docs/EXPLANATION_CHOOSE_A_SETUP.md]:
```suggestion
After you have [decided Spoke is right for you](EXPLANATION_DECIDING_ON_SPOKE.md), you will need to decide how you want to deploy.
```

File: README.md [hunk 8]
```diff
@@ -34,121 +34,46 @@ Please let us know if you deployed by filling out this form [here](https://act.m
    ```
 2. Install yarn.
```
Comment (sharonsolomon) [inline README.md]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -9,7 +9,7 @@ Spoke was created by Saikat Chakrabarti and Sheena Pakanati, and is now maintain
 
 The latest version is [9.0](https://github.com/MoveOnOrg/Spoke/tree/v9.0) (see [release notes](https://github.com/MoveOnOrg/Spoke/blob/main/docs/RELEASE_NOTES.md#v90))
 
-## Deploy to Heroku
+
 
 Use the Heroku Button to deploy a version of Spoke suitable for testing. This won't cost any money and will not support production usage. It's a great way to practice deploying Spoke or see it in action.
```
Comment (sharonsolomon) [inline README.md]:
```suggestion
```

Comment (sharonsolomon) [inline README.md]:
```suggestion
```

**Comment by ibrand** [comment]
I saw you mention the microsite link in a few areas, here it is! https://moveonorg.github.io/Spoke/#/

File: docs/HOWTO_DEVELOPMENT_LOCAL_SETUP.md (hunk 2 above)
Comment (sharonsolomon) [reply docs/HOWTO_DEVELOPMENT_LOCAL_SETUP.md]: Deleted the paragraph here. Fixed in readme in commit a5ff9d6.

**Review by ibrand — APPROVED** [review]
Thank you for your iterations. This is a huge improvement to the old README! 🥳

**Pull request merged**